File: ros/ros_packages/rosagent/rosagent/message/ingress_system_message.py
class IngressSystemMessage:

    def __init__(self, metadata, data):
        self._metadata = metadata
        self._data = data

    def get_metadata(self):
        return self._metadata

    def get_data(self):
        return self._data

    def get_node_type(self):
        return self._metadata.get_node_type()
    
    def get_node_name(self):
        return self._metadata.get_node_name()

    def get_request_id(self):
        return self._metadata.get_request_id()

    def get_message_manager(self):
        return self._metadata.get_message_manager()

    def is_management_message(self):
        return self._metadata.has_message_manager()

    def is_interface_node_message(self):
        return self._metadata.has_node_name()

File: unity/Assets/Scripts/unityagent/SystemMessage/IngressSystemMessage.cs
namespace UnityAgent.SystemMessage
{
public class IngressSystemMessage
{
    private readonly SystemMessageMetadata metadata;
    private readonly byte[] data;

    public IngressSystemMessage(SystemMessageMetadata metadata, byte[] data)
    {
        this.metadata = metadata;
        this.data = data;
    }

    public SystemMessageMetadata Metadata => metadata;
    public byte[] Data => data;
    public byte SenderAgentType => metadata.SenderAgentType;
    public ushort SenderAgentId => metadata.SenderAgentId;
    public byte NodeType => metadata.NodeType;
    public string NodeName => metadata.NodeName;
    public uint RequestId => metadata.RequestId;
    public string MessageManager => metadata.MessageManager;
    
    public bool IsManagementMessage()
    {
        return metadata.HasMessageManager();
    }

    public bool IsInterfaceNodeMessage()
    {
        return metadata.HasNodeName();
    }
}
}